src/librustc/hir/lowering.rs

```diff
@@ -1833,8 +1835,12 @@ impl<'a> LoweringContext<'a> {
                 ExprKind::Try(ref sub_expr) => {
                     // to:
                     //
+                    // #[allow(unreachable_patterns)]
```

Comment:
The `unreachable_code` attribute is indeed fine - no user-defined code exists under it. It's the `unreachable_patterns` attribute that is the problem - could you check why is it needed?